pulsar-broker/src/main/java/com/yahoo/pulsar/broker/loadbalance/BrokerHostUsage.java

```diff
@@ -71,25 +91,152 @@ public BrokerHostUsage(PulsarService pulsar) {
      *
      * @throws IOException
      */
-    public String getBrokerHostUsage() throws IOException {
-        StringWriter writer = new StringWriter();
+    public synchronized SystemResourceUsage getBrokerHostUsage() throws IOException {
```

Comment:
The refresh should not be done every time this method is called, but rather at regular intervals (eg: every 1min)